modules/PathUtils.js

```diff
@@ -37,16 +37,32 @@ export const parsePath = (path) => {
   }
 }
 
+const validatePath = (path) => {
+  try {
+    decodeURI(path)
+  } catch(e) {
+    if (e instanceof URIError) {
+      throw new Error(
+        'You are attempting to create a path that cannot be decoded. ' +
+        'Please call encodeURI on the pathname.'
+      )
```

Comment:
A slightly more useful error than the default `Uncaught URIError: URI malformed`